canonicalwebteam/discourse_docs/blueprint.py

```diff
@@ -0,0 +1,66 @@
+import flask
+from requests.exceptions import HTTPError
+
+from canonicalwebteam.discourse_docs.models import (
+    NavigationParseError,
+    RedirectFoundError,
+)
+
+
+def build_blueprint(url_prefix, model):
+
+    blueprint = flask.Blueprint("discourse_docs", __name__)
+
+    @blueprint.errorhandler(404)
+    def page_not_found(e):
+        try:
+            frontpage, nav_html = model.parse_frontpage()
+        except NavigationParseError as nav_error:
+            nav_html = f"<p>{str(nav_error)}</p>"
+
+        return flask.render_template("docs/404.html", nav_html=nav_html), 404
```

Comment:
Do you think we should still be providing a custom 404 page just so we can include the docs nav on the left? I think maybe we should keep it simple and just throw the 404 back to the application to be handled with its standard error handler?

Reply:
I don't really know what is best, maybe yeah we could let the app handle it to avoid having different behaviours depending on the page we are.